### DISCUSSION TITLE
Add support for themes

### DISCUSSION THREAD

**Discussion by iostreamer-X** [metz-sh/simulacrum#38] [Ideas]
#36

**Comment by iostreamer-X** [comment]
Some caveats though:
- The theme.json which monaco(the editor uses) is pretty straightforward. Ideally we would want end-to-end theming, where users can select from a bunch. But let’s start with simple “light” & “dark”.
- With IDE handled, then comes the rest of the application. That’d be the file-tree, the status-console, and other surrounding bits. They all need a “light” version so that the entire application looks cohesive. We wouldn’t want only parts of it to be “light”.

For example, here are the two theme files:
https://github.com/metz-sh/simulacrum/blob/main/src/ui/components/story-script-console/theme.json
https://github.com/metz-sh/simulacrum/blob/main/src/ui/components/ide/theme.json

Let’s start with the simplest part, i.e. just a “light” theme for editor. Then we can move on.

**Comment by iostreamer-X** [comment]
> Before I start coding, I’d like to confirm if I’ve understood the task correctly and if my proposed approach aligns with your expectations. Please feel free to correct me if I’m off track anywhere.

>I plan to modify theme.json as :
{   "dark": { "base": "vs-dark", "inherit": true, // others ...   },   "light": { "base": "vs", "inherit": true, // others ....   } }

> Then, I’ll add a button in monaco-editor.tsx to toggle the theme. I’ll use either a Zustand store, local storage, or session storage to persist the theme selection across page reloads or revisits.


Let's have themes in different files rather than stuffing them all in one `.json` file. This will enable us to load only what we need and offer better modularity.

Secondly, zustand is already integrated, let's use that. We can then have zustand sync to localstorage for us.

**Comment by SahilLamba0008** [comment]
okay,  I've a few additional questions : 
1. Should the theme change be applied only to the Editor component, or do you expect it to affect the entire application's theme, including the Playground component and other parts as well?
2. Where do you suggest I add the theme toggle button? Should it be placed within the Editor component itself, or would it be better suited elsewhere in the application?

**Reply by iostreamer-X** [reply]
1. Let’s start simple and add it only to editor and in the next stage we will be adding across the app.
2. Feel free to choose! Your call. We can iterate if it doesn’t look good.
